src/storage/dbworker.cpp

```diff
@@ -550,7 +550,7 @@ void DBWorker::getHistory(const QString &filter)
 
     if (!filter.isEmpty()) {
         filterQuery = filterQuery.arg(QString("(url LIKE :search OR title LIKE :search)"));
-        order = QString("LENGTH(url), title, date ASC");
+        order = QString("visited_count desc, LENGTH(url), title, date ASC");
```

Comment:
It's important that sites rankings decay if they haven't been visited for some time. Maybe bring 'date' up to the second place?